core/src/main/java/hudson/PluginManager.java

```diff
@@ -928,56 +928,68 @@ public void dynamicLoad(File arc, boolean removeExisting) throws IOException, In
                 p.resolvePluginDependencies();
                 strategy.load(p);
 
-                Jenkins.get().refreshExtensions();
+                if (batch != null) {
+                    batch.add(p);
+                } else {
+                    start(Collections.singletonList(p));
+                }
 
-                p.getPlugin().postInitialize();
             } catch (Exception e) {
                 failedPlugins.add(new FailedPlugin(sn, e));
                 activePlugins.remove(p);
                 plugins.remove(p);
                 throw new IOException("Failed to install "+ sn +" plugin",e);
             }
 
-            // run initializers in the added plugin
-            Reactor r = new Reactor(InitMilestone.ordering());
-            final ClassLoader loader = p.classLoader;
-            r.addAll(new InitializerFinder(loader) {
-                @Override
-                protected boolean filter(Method e) {
-                    return e.getDeclaringClass().getClassLoader() != loader || super.filter(e);
-                }
-            }.discoverTasks(r));
-            try {
-                new InitReactorRunner().run(r);
-            } catch (ReactorException e) {
-                throw new IOException("Failed to initialize "+ sn +" plugin",e);
+            LOGGER.log(INFO, "Plugin {0}:{1} dynamically {2}", new Object[] {p.getShortName(), p.getVersion(), batch != null ? "loaded but not yet started" : "installed"});
```

Comment:
At the moment each plugin that gets installed gets 5 lines of logs.

```
juil. 24, 2019 3:08:24 PM hudson.model.UpdateCenter$DownloadJob run
INFOS: Starting the installation of Trilead API on behalf of anonymous
juil. 24, 2019 3:08:25 PM hudson.model.UpdateCenter$UpdateCenterConfiguration download
INFOS: Downloading Trilead API
juil. 24, 2019 3:08:27 PM hudson.PluginManager dynamicLoad
INFOS: Attempting to dynamic load /tmp/jenkins/war/work/plugins/trilead-api.jpi
juil. 24, 2019 3:08:27 PM hudson.PluginManager dynamicLoad
INFOS: Plugin trilead-api:1.0.4 dynamically loaded but not yet started
juil. 24, 2019 3:08:27 PM hudson.model.UpdateCenter$DownloadJob run
INFOS: Installation successful: Trilead API
```

I think this message should be on level `FINE`, as well as `Attempting to dynamic load....`. Additionally, I think the plugin display name, not the short name, should be used to be consistent with the rest of the plugin installation logging.

Reply:
> I think this message should be on level `FINE`, as well as `Attempting to dynamic load....`.

Probably a good idea.

> I think the plugin display name, not the short name, should be used to be consistent with the rest of the plugin installation logging.

To the contrary, I think short names should always be used in log files. Display names are for the GUI.

Reply:
> To the contrary, I think short names should always be used in log files. Display names are for the GUI.

As long as it is consistent, I'm fine with it.